### PR TITLE
Fix issue #86: checking default values

### PR DESCRIPTION
**Pull request by bwbeach**
See the comments in `B2JsonHandlerMap` for the new approach to
checking default values.

Testing: unit tests

### CODE REVIEW

File: core/src/main/java/com/backblaze/b2/json/B2JsonHandlerMap.java [hunk 1]
```diff
@@ -172,13 +181,41 @@ private B2JsonHandlerMap(Map<Class<?>, B2JsonTypeHandler<?>> initialMapOrNull) {
             throw new B2JsonException(t.getMessage());
         }
         finally {
+            // Remember the handlers we added so we can check their defaults.
+            handlersToCheckDefaults = new ArrayList<>(handlersAddedToMap);
```
[the remark this answers is not in the document]
Comment (bwbeach) [reply core/src/main/java/com/backblaze/b2/json/B2JsonHandlerMap.java]: Yep.

File: core/src/main/java/com/backblaze/b2/json/B2JsonHandlerMap.java [hunk 2]
```diff
@@ -121,6 +121,12 @@ private B2JsonHandlerMap(Map<Class<?>, B2JsonTypeHandler<?>> initialMapOrNull) {
 
         // This method is NOT re-entrant.  The code that creates and initializes new handlers
         // should not call this method.
+        //
+        // The reason this code cannot be re-entrant is that it would try to create more
+        // new handlers, which could make it impossible for this method to un-do what it
+        // has done by removing the handlers it had added.  A re-entrant call could
+        // wind up creating dependencies on the classes we temporarily added but wound
```
[the remark this answers is not in the document]
Comment (bwbeach) [reply core/src/main/java/com/backblaze/b2/json/B2JsonHandlerMap.java]: Stickin with "wound up removing".

[the remark this answers is not in the document]
Comment (certainmagic) [reply core/src/main/java/com/backblaze/b2/json/B2JsonHandlerMap.java]: :)

File: core/src/main/java/com/backblaze/b2/json/B2JsonHandlerMap.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (certainmagic) [reply core/src/main/java/com/backblaze/b2/json/B2JsonHandlerMap.java]: :)

**Pull request merged**